src/storage/raw/raw_basic.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use api_version::{APIV2, APIVersion};
+use crate::storage::kv::{Iterator, Result, Snapshot};
+
+use engine_traits::{CfName, DATA_KEY_PREFIX_LEN};
+use engine_traits::{IterOptions, ReadOptions};
+use tikv_kv::Modify;
+use txn_types::{Key, TimeStamp, Value};
+
+#[derive(Clone)]
+pub struct RawBasicSnapshot<S: Snapshot> {
+    snap: S,
+}
+
+impl<S: Snapshot> RawBasicSnapshot<S> {
+    pub fn from_snapshot(snap: S) -> Self {
+        RawBasicSnapshot { snap }
+    }
+
+    pub fn seek_first_key_value_cf(
```

Comment:
If just `seek_first_key_value_cf` is needed, maybe it is too verbose to have a new snap. We can extract this method, and invoke directly on `engine`'s snapshot.